section/models.py

```diff
@@ -303,9 +303,12 @@ def get_context(self, request, *args, **kwargs):
         return context
     
     def get_section_articles(self, order='-explicit_published_at') -> QuerySet:
+        import datetime
         # return ArticlePage.objects.from_section(section_root=self)
         # section_articles = ArticlePage.objects.live().public().filter(current_section=self.slug).order_by(order)
-        section_articles = ArticlePage.objects.live().public().descendant_of(self).order_by(order)
+        #section_articles = ArticlePage.objects.live().public().descendant_of(self).order_by(order)
+        section_articles = ArticlePage.objects.live().child_of(self).filter(explicit_published_at__gt=datetime.datetime.now()-datetime.timedelta(days=60)).order_by(order)
+        
```

Comment:
I noticed that this queryset no longer includes the `.public()` selector, which AFAIK was there to prevent private pages from showing up on the homepage.

When profiling the site I discovered that the `.public()` selector slows queries down quite a bit, I think because Wagtail doesn't create a database index for public/private pages (`.live()`, on the other hand, is indexed). I recommend adding `.public()` back and remeasuring the query times. It could be that most of your speed up is coming from the removal of this call rather than the new filter. If that's the case, unfortunately I don't think it's worth it to add the complexity of the 60 day filter.

I've been wondering if we could get away without the `.public()` selector if private pages aren't being used, but wanted to play it safe and explore that optimization later.

Reply:
I think you are right. It seemed to reduce queries (269->260) but I think the load time was actualy increased a bit.

We don't have any pages deeper than the section pages so I don't beleive we need `.public()`